deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/SpecConfig.java

```diff
@@ -1,35 +1,29 @@
 package io.quarkiverse.openapi.generator.deployment;
 
 import java.nio.file.Path;
-import java.util.Collections;
 import java.util.Map;
 
-import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
 
 // This configuration is read in codegen phase (before build time), the annotation is for document purposes and avoiding quarkus warns
-@ConfigGroup
-@ConfigRoot(prefix = SpecConfig.BUILD_TIME_CONFIG_PREFIX)
+@ConfigRoot(name = SpecConfig.BUILD_TIME_CONFIG_PREFIX, phase = ConfigPhase.BUILD_TIME)
 public class SpecConfig {
 
-    public static final String BUILD_TIME_CONFIG_PREFIX = "quarkus.openapi-generator.codegen";
+    static final String BUILD_TIME_CONFIG_PREFIX = "openapi-generator.codegen";
     public static final String API_PKG_SUFFIX = ".api";
     public static final String MODEL_PKG_SUFFIX = ".model";
     // package visibility for unit tests
-    static final String BUILD_TIME_SPEC_PREFIX_FORMAT = BUILD_TIME_CONFIG_PREFIX + ".spec.\"%s\"";
+    static final String BUILD_TIME_SPEC_PREFIX_FORMAT = "quarkus." + BUILD_TIME_CONFIG_PREFIX + ".spec.\"%s\"";
     private static final String BASE_PACKAGE_PROP_FORMAT = "%s.base-package";
     private static final String SKIP_FORM_MODEL_PROP_FORMAT = "%s.skip-form-model";
 
     /**
      * OpenAPI Spec details for codegen configuration.
      */
     @ConfigItem(name = "spec")
-    Map<String, SpecItemConfig> specItem;
-
-    public Map<String, SpecItemConfig> getSpecItem() {
-        return Collections.unmodifiableMap(specItem);
-    }
+    public Map<String, SpecItemConfig> specItem;
```

Comment:
If this public field ever used?, I cannot find any code using it

Reply:
It's for document purposes only and Quarkus bind this property, so it must be public.